sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java

```diff
@@ -0,0 +1,164 @@
+            Optional.ofNullable(readTimeoutMillis).ifPresent(options::setReadTimeoutMillis);
+            Optional.ofNullable(sampleRate).ifPresent(options::setSampleRate);
+            Optional.ofNullable(bypassSecurity).ifPresent(options::setBypassSecurity);
+            Optional.ofNullable(debug).ifPresent(options::setDebug);
+            Optional.ofNullable(attachThreads).ifPresent(options::setAttachThreads);
+            Optional.ofNullable(attachStacktrace).ifPresent(options::setAttachStacktrace);
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  protected void append(@NotNull ILoggingEvent eventObject) {
+    if (Sentry.isEnabled()) {
+      Sentry.captureEvent(createEvent(eventObject));
+    }
+  }
+
+  /**
+   * Creates {@link SentryEvent} from Logback's {@link ILoggingEvent}.
+   *
+   * @param loggingEvent the logback event
+   * @return the sentry event
+   */
+  @SuppressWarnings("JdkObsolete")
+  final @NotNull SentryEvent createEvent(@NotNull ILoggingEvent loggingEvent) {
+    final SentryEvent event = new SentryEvent(new Date(loggingEvent.getTimeStamp()));
+    final Message message = new Message();
+    message.setFormatted(loggingEvent.getFormattedMessage());
+    message.setParams(toParams(loggingEvent.getArgumentArray()));
+    event.setMessage(message);
+    event.setLogger(loggingEvent.getLoggerName());
+    event.setLevel(formatLevel(loggingEvent.getLevel()));
+
+    final ThrowableProxy throwableInformation = (ThrowableProxy) loggingEvent.getThrowableProxy();
+    if (throwableInformation != null) {
+      event.setThrowable(throwableInformation.getThrowable());
+    }
+
+    if (loggingEvent.getThreadName() != null) {
+      event.setExtra("thread_name", loggingEvent.getThreadName());
+    }
+
+    for (Map.Entry<String, String> entry : loggingEvent.getMDCPropertyMap().entrySet()) {
+      event.setTag(entry.getKey(), entry.getValue());
+    }
+
+    return event;
+  }
+
+  private @NotNull List<String> toParams(@Nullable Object[] arguments) {
+    if (arguments != null) {
+      return Arrays.stream(arguments)
+          .filter(Objects::nonNull)
+          .map(Object::toString)
+          .collect(Collectors.toList());
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  /**
+   * Transforms a {@link Level} into an {@link SentryLevel}.
+   *
+   * @param level original level as defined in log4j.
+   * @return log level used within sentry.
+   */
+  private static @NotNull SentryLevel formatLevel(@NotNull Level level) {
+    if (level.isGreaterOrEqual(Level.ERROR)) {
+      return SentryLevel.ERROR;
+    } else if (level.isGreaterOrEqual(Level.WARN)) {
+      return SentryLevel.WARNING;
+    } else if (level.isGreaterOrEqual(Level.INFO)) {
+      return SentryLevel.INFO;
+    } else if (level.isGreaterOrEqual(Level.ALL)) {
+      return SentryLevel.DEBUG;
+    } else {
+      return SentryLevel.LOG;
+    }
```

Comment:
a `switch` here would not be cheaper? but fine with it.

I'd avoid `SentryLevel.LOG` at all, this is just for compatibility with sentry-react-native, so in case you need to fallback, do it with `DEBUG`

Reply:
`LOG` removed. I just realised that `level` here is not an enum, we cannot make a switch.